testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/elytron/seccontext/ElytronClientDefaultSSLContextProviderTestCase.java

```diff
@@ -0,0 +1,254 @@
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.security.auth.x500.X500Principal;
+import javax.ws.rs.client.ClientBuilder;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.security.KeyManagementException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Security;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Test that the ElytronClientDefaultSSLContextProvider can be used to configure SSL context for RESTEasy client by calling SSLContext.getDefault()
+ */
+public class ElytronClientDefaultSSLContextProviderTestCase {
+
+    private static final String CONFIG_FILE = "file:./src/test/resources/wildfly-config-default-ssl-context.xml";
+    private static final String CLIENT_KEYSTORE = "client.keystore";
+    private static final String SERVER_KEYSTORE = "server.keystore";
+    private static final String CLIENT_TRUSTSTORE = "client.truststore";
+    private static final String SERVER_TRUSTSTORE = "server.truststore";
+    private static final String KEYSTORE_TYPE = "JKS";
+    private static final char[] PASSWORD = "password".toCharArray();
+    private static final String CLIENT_ALIAS = "client";
+    private static final String LOCALHOST_ALIAS = "localhost";
+    private static final String RSA = "RSA";
+    private static final String SHA_256_WITH_RSA = "SHA256withRSA";
+    private static File KEYSTORES_DIR = new File("./target/keystores");
+    private int port = 10000;
+    private AtomicReference<Boolean> running = new AtomicReference<>(false);
+    private static SSLSocket connectionSocket = null;
+    private static ServerSocket sslServerSocket;
+
+    @BeforeClass
+    public static void generateKeystoreAndConfigureProvider() throws CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        if (KEYSTORES_DIR.exists()) {
+            new File(KEYSTORES_DIR, CLIENT_KEYSTORE).delete();
+            new File(KEYSTORES_DIR, SERVER_KEYSTORE).delete();
+            new File(KEYSTORES_DIR, CLIENT_TRUSTSTORE).delete();
+            new File(KEYSTORES_DIR, SERVER_TRUSTSTORE).delete();
+            KEYSTORES_DIR.delete();
+        }
+        generateTwoWaySSLKeyStoresAndTrustStores(new File(KEYSTORES_DIR, CLIENT_KEYSTORE).getPath(), new File(KEYSTORES_DIR, SERVER_KEYSTORE).getPath(),
```

Comment:
I am wondering whether we could have some `private final File *_*STORE _FILE` replacing all of these `new File(...` occurrences for better readability.

Reply:
@fabiobrz This is no longer present